Repo restruct

akomii opened the pull request:
- Refactoring of python-ql Github action
- Restructuring of repos folder structure to hold now Github actions and Github workflows

akomii committed:
refactored python ql action to be more customizable, updated dependent actions version

akomii committed:
moved python ql action into subfolder

akomii committed:
Updated Readme to changed folder structure (division into actions and workflows). Added generic usage example of custom github workflows

bunbry commented on actions/python-ql/action.yml:
```diff
@@ -37,18 +52,22 @@ runs:
+        path: ${{ env.pip-cache-dir }}
+        key: ${{ runner.os }}-pip-${{ github.run_id }}
         restore-keys: |
           ${{ runner.os }}-pip-
```
The setup-python action (like all actions/setup-xyz) have integrated caching for dependencies. More on that here: https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies

bunbry commented on actions/python-ql/action.yml:
```diff
@@ -87,7 +106,8 @@ runs:
       run: |
-        if [[ -f "${{ inputs.integration-test-path }}" ]]; then
+        set -e
+        if [[ -n "${{ inputs.integration-test-path }}" && -f "${{ inputs.integration-test-path }}" ]]; then
```
Is there a reason to use  `-n && -f` instead of just `-f` as condition? Occurs multiple times in file.

bunbry reviewed (changes requested):
The caching should be updated to use the integrated configuration of setup-python action. 
If there is no specific reason for the if-condition it should be briefly discussed which version will be used.

akomii committed:
adjustments to code review:
- removed redundant -n check (is already done by -f)
- removed custom package caching as its already done by setup-python action
